src/lib.rs

```diff
@@ -2096,6 +2266,324 @@ where
     }
 }
 
+///
+/// Common methods shared by all OAuth2 token inspection implementations.
+///
+/// The methods in this trait are defined in
+/// [Section 2.2 of RFC 7662](https://tools.ietf.org/html/rfc7662#section-2.2). This trait exists
+/// separately from the `StandardTokenInspectionResponse` struct to support customization by clients,
+/// such as supporting interoperability with non-standards-complaint OAuth2 providers.
+///
+pub trait TokenInspectionResponse<TT>: Debug + DeserializeOwned + Serialize
+where
+    TT: TokenType,
+{
+    ///
+    /// REQUIRED.  Boolean indicator of whether or not the presented token
+    /// is currently active.  The specifics of a token's "active" state
+    /// will vary depending on the implementation of the authorization
+    /// server and the information it keeps about its tokens, but a "true"
+    /// value return for the "active" property will generally indicate
+    /// that a given token has been issued by this authorization server,
+    /// has not been revoked by the resource owner, and is within its
+    /// given time window of validity (e.g., after its issuance time and
+    /// before its expiration time).
+    ///
+    fn active(&self) -> bool;
+    ///
+    ///
+    /// OPTIONAL.  A JSON string containing a space-separated list of
+    /// scopes associated with this token, in the format described in
+    /// [Section 3.3 of OAuth 2.0](https://tools.ietf.org/html/rfc7662#section-3.3).
+    /// If included in the response,
+    /// this space-delimited field is parsed into a `Vec` of individual scopes. If omitted from
+    /// the response, this field is `None`.
+    ///
+    fn scopes(&self) -> Option<&Vec<Scope>>;
+    ///
+    /// OPTIONAL.  Client identifier for the OAuth 2.0 client that
+    /// requested this token.
+    ///
+    fn client_id(&self) -> Option<&ClientId>;
+    ///
+    /// OPTIONAL.  Human-readable identifier for the resource owner who
+    /// authorized this token.
+    ///
+    fn username(&self) -> Option<&String>;
```

Comment:
let's return `&str` instead of `&String` for these, which is a bit more idiomatic in Rust I think. I used `&String` in a few places from when I was less familiar with the Rust conventions

Reply:
I've updated all but `fn aud(&self) -> Option<&Vec<String>>;` since it stores all audience strings as a string internally